doc/api/esm.md

```diff
@@ -348,36 +351,35 @@ changes:
 
 * `specifier` {string} The module specifier to resolve relative to the
   current module.
-* Returns: {string} The absolute (`file:`) URL string for the resolved module.
+* Returns: {string} The absolute URL string that the specifier would resolve to.
 
 [`import.meta.resolve`][] is a module-relative resolution function scoped to
 each module, returning the URL string.
 
 ```js
 const dependencyAsset = import.meta.resolve('component-lib/asset.css');
 // file:///app/node_modules/component-lib/asset.css
+import.meta.resolve('./dep');
+// file:///app/dep
 ```
 
 All features of the Node.js module resolution are supported. Dependency
 resolutions are subject to the permitted exports resolutions within the package.
 
```

Comment:
Maybe add an example of resolving relative to a different base by passing in a URL instance? (On my phone or I'd try to write one.)

Reply:
Do you mean `import.meta.resolve(new URL('./whatever', 'http://example.com/'))`? Why would anyone do that? It's equivalent to `new URL('./whatever', 'http://example.com/').href` except it needs to cross threads. Or do you mean something else?

Reply:
> Do you mean `import.meta.resolve(new URL('./whatever', 'http://example.com/'))`? Why would anyone do that?

Yes, that’s what I meant. The reason someone would do that is because it’s a workaround for the lack of the `parentURL` second parameter (in the stable version of the API). Though you raise a good point, I can’t think of when it would be better than just `new URL`, other than if you want to ensure that any custom `resolve` hooks get called (like if you had a custom hook to transform `file.ts` to `file.js` or something, say). Not sure if that’s a common enough use case to be worth documenting though.